components/FolderNav.tsx

```diff
@@ -0,0 +1,171 @@
+import { styled } from "styled-components";
+import Image from "next/image";
+import { UserFolder } from "@/lib/Type";
+import {
+  Dispatch,
+  MouseEvent,
+  SetStateAction,
+  SyntheticEvent,
+  useState,
+} from "react";
+import { useSearchParams } from "next/navigation";
+
+interface FolderNavProps {
+  folders: UserFolder[];
+  setCurrentFolderID: Dispatch<SetStateAction<number>>;
+}
+
+function FolderNav({ folders = [], setCurrentFolderID }: FolderNavProps) {
+  const [currentFolder, setCurrentFolder] = useState({ name: "전체", id: 0 });
+
+  const handleClick = (name: string, id: number) => {
+    console.log(id);
+    setCurrentFolderID(id);
+    setCurrentFolder({ name: name, id: id });
+  };
+
+  return (
+    <Container>
+      <ButtonContainer>
+        <Buttons>
+          <FolderButton
+            key={0}
+            isClicked={currentFolder.id === 0}
+            onClick={() => {
+              handleClick("전체", 0);
+            }}
+          >
+            전체
+          </FolderButton>
+          {folders &&
+            folders.map((folder) => {
+              return (
+                <FolderButton
+                  key={folder.id}
+                  isClicked={currentFolder.id === folder.id}
+                  onClick={() => {
+                    handleClick(folder.name, folder.id);
+                  }}
+                >
+                  {folder.name}
+                </FolderButton>
+              );
+            })}
+        </Buttons>
+        <FolderAddButton>폴더 추가 +</FolderAddButton>
+      </ButtonContainer>
+      <Div>
+        <FolderName>{currentFolder.name}</FolderName>
+        <CTAs>
+          <CTA>
+            <S_Image src={"/images/share.svg"} width={18} height={18} alt="" />
+            공유
+          </CTA>
+          <CTA>
+            <S_Image src={"/images/pen.svg"} width={18} height={18} alt="" />
+            이름 변경
+          </CTA>
+          <CTA>
+            <S_Image src={"/images/delete.svg"} width={18} height={18} alt="" />
+            삭제
+          </CTA>
+        </CTAs>
+      </Div>
+      <NoLinksFound></NoLinksFound>
+    </Container>
+  );
+}
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  max-width: 106rem;
+  margin: 0 auto;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin: 4rem 0;
+`;
+
+const Buttons = styled.div`
+  display: flex;
+`;
+
+const FolderButton = styled.button<{ isClicked: boolean }>`
+  box-sizing: border-box;
+  display: inline-block;
+
+  min-width: 5rem;
+  width: fit-content;
+  max-width: 15rem;
+
+  overflow: hidden;
+  cursor: pointer;
+
+  height: 3.5rem;
+  padding: 0.8rem 1.2rem;
+  margin-right: 0.8rem;
+
+  border-radius: 5px;
+  border: 1px solid var(--linkbrary-primary-color, #6d6afe);
+  background-color: ${({ isClicked }) => (isClicked ? "#6d6afe" : "#ffffff")};
```

Comment:
요런것도 마이너한 코멘트긴 한데 styled components의 css 모듈 활용하시면,
여러번 props 받아올 필요없이 한번에 스타일링 해줄수있답니다 :)

```suggestion
  ${({ isClicked }) => css`
    background-color: ${isClicked ? "#6d6afe" : "#ffffff"};
    color: ${isClicked ? "#ffffff" : "#000000"};
  `}
```